internal/util/update/resource-merge.go

```diff
@@ -110,3 +119,130 @@ func (u ResourceMergeUpdater) updatedKptfile(updatedPath string, options UpdateO
 	err = kf.MergeOpenAPI(options.KptFile)
 	return kf, err
 }
+
+// replaceNonKRMFiles replaces the non KRM files in localDir with the corresponding files in updatedDir,
+// it also deletes non KRM files and sub dirs which are present in localDir and not in updatedDir
+func ReplaceNonKRMFiles(updatedDir, originalDir, localDir string) error {
+	updatedSubDirs, updatedFiles, err := getSubDirsAndNonKrmFiles(updatedDir)
+	if err != nil {
+		return err
+	}
+
+	originalSubDirs, originalFiles, err := getSubDirsAndNonKrmFiles(originalDir)
+	if err != nil {
+		return err
+	}
+
+	localSubDirs, localFiles, err := getSubDirsAndNonKrmFiles(localDir)
+	if err != nil {
+		return err
+	}
+
+	// identify all non KRM files modified locally, to leave them untouched
+	locallyModifiedFiles := sets.String{}
+	for _, file := range localFiles.List() {
+		if !originalFiles.Has(file) {
+			// new local file has been added
+			locallyModifiedFiles.Insert(file)
+			continue
+		}
+		same, err := compareFiles(filepath.Join(originalDir, file), filepath.Join(localDir, file))
+		if err != nil {
+			return err
+		}
+		if !same {
```

Comment:
consider updating only if it has been modified in the update from the original to reduce file writes.

Reply:
Will do.

Reply:
Actually I will do this in SyncFile util method in kyaml. I can do it as a sub item in my next PR.